### ISSUE TITLE
[FEA] create a formal process for updateing the github-pages branch

### ISSUE THREAD

**Issue by revans2** [NVIDIA/spark-rapids#471]
Right now we have docs in the release branch and docs in the github-pages branch.  We don't really have any separation between docs that are associated with a release and docs that are general.  This can lead to issues like here

https://github.com/NVIDIA/spark-rapids/pull/467#discussion_r463029568

where we try to update general documentation and end up updating release specific information.

We need a formal split between the docs, or we need a good process in place to avoid this in the future.

**Comment by revans2** [comment]
We discussed this and for now only a hot fix can go into gh-pages directly and in parallel go into the current development branch.  In all other cases the changes should only go into the current development branch, and it will be merged into gh-pages as a part of the release.

At some point when we are supporting more then one release at a time we can look at how to have version specific docs in gh-pages.
